File: 0101-symmetric-tree/0101-symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    function solve(left, right){
        if (left==null && right==null){
            return true
        }
        if(left==null || right==null){
            return false
        }
        return (left.val == right.val && solve(left.left, right.right) && solve(left.right, right.left))
     
        

    }
    if (root==null)
    return true

    return solve(root.left, root.right);
    
};

File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool solve(TreeNode* l, TreeNode* r){
        if(l==NULL && r==NULL){
            return true;
        }
        if(l==NULL ||  r==NULL){
            return false;
        }
        return (l->val ==r->val && solve(l->left, r->right) && solve(l->right, r->left));
    }
    bool isSymmetric(TreeNode* root) {
        if (root==NULL){
            return true;
        }
        TreeNode* l = root->left;
        TreeNode* r = root->right;
        return solve(l,r);
        
    }
};